src/features/broker-protection/utils.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Get a single element.
+ *
+ * @param {HTMLElement | Element | Node | Document} doc
+ * @param {string} selector
+ * @return {HTMLElement | null}
+ */
+export function getElement (doc = document, selector) {
+    if (isXpath(selector)) {
+        return safeQuerySelectorXPath(doc, selector)
+    }
+
+    return safeQuerySelector(doc, selector)
+}
+
+/**
+ * Get an array of elements
+ *
+ * @param {HTMLElement | Element | Node | Document} doc
+ * @param {string} selector
+ * @return {HTMLElement[] | null}
+ */
+export function getElements (doc = document, selector) {
+    if (isXpath(selector)) {
+        return safeQuerySelectorAllXpath(doc, selector)
+    }
+
+    return safeQuerySelectorAll(doc, selector)
+}
+
+/**
+ * Test if a given selector matches an element.
+ *
+ * @param {HTMLElement} element
+ * @param {string} selector
+ */
+export function getElementMatches (element, selector) {
+    try {
+        if (isXpath(selector)) {
+            return matchesXPath(element, selector) ? element : null
+        } else {
+            return element.matches(selector) ? element : null
+        }
+    } catch (e) {
+        console.error('getElementMatches threw: ', e)
+        return null
+    }
+}
+
+/**
+ * This is a xpath version of `element.matches(CSS_SELECTOR)`
+ * @param {HTMLElement} element
+ * @param {string} selector
+ * @return {boolean}
+ */
+function matchesXPath (element, selector) {
+    const xpathResult = document.evaluate(
+        selector,
+        element,
+        null,
+        XPathResult.BOOLEAN_TYPE,
+        null
+    )
+
+    return xpathResult.booleanValue
+}
+
+/**
+ * @param {unknown} selector
+ * @returns {boolean}
+ */
+function isXpath (selector) {
+    if (!(typeof selector === 'string')) return false
+    return selector.startsWith('//') || selector.startsWith('./') || selector.startsWith('(')
+}
+
+/**
+ * @param {HTMLElement|Element|Node|Document} element
+ * @param selector
+ * @returns {HTMLElement[] | null}
+ */
+function safeQuerySelectorAll (element, selector) {
+    try {
+        if (element && 'querySelectorAll' in element) {
+            return Array.from(element?.querySelectorAll?.(selector))
+        }
+        return null
+    } catch (e) {
+        return null
+    }
+}
+/**
+ * @param {HTMLElement|Element|Node|Document} element
```

Comment:
wouldn't just `Node` be enough? Also for other functions here

Reply:
I'll try - I think it might be related to splitting the PRs up, I'm sure I widened this to fix something.

Reply:
Done!